internal/component/azure/eventhubs/eventhubs.go

```diff
@@ -138,11 +152,94 @@ func (aeh *AzureEventHubs) Publish(ctx context.Context, topic string, messages [
 	return nil
 }
 
+// GetPubSubHandlerFunc returns the handler function for pubsub messages
+func (aeh *AzureEventHubs) GetBindingsHandlerFunc(topic string, getAllProperties bool, handler bindings.Handler, timeout time.Duration) HandlerFn {
+	return func(ctx context.Context, messages []*azeventhubs.ReceivedEventData) ([]HandlerResponseItem, error) {
+		if len(messages) != 1 {
+			return nil, fmt.Errorf("expected 1 message, got %d", len(messages))
+		}
+
+		bindingsMsg, err := NewBindingsReadResponseFromEventData(messages[0], topic, getAllProperties)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get bindings read response from azure eventhubs message: %+v", err)
```

Comment:
For errors in `fmt.Errorf` we use `%w` (but we use `%v` in log statements)
```suggestion
			return nil, fmt.Errorf("failed to get bindings read response from azure eventhubs message: %w", err)
```